nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/hosted/HelmHostedFacetImpl.java

```diff
@@ -95,31 +94,33 @@ public Content get(final String path) {
   }
 
   @Override
+  @TransactionalStoreBlob
   public void upload(final String path,
                      final Payload payload,
                      final AssetKind assetKind) throws IOException
   {
-    checkNotNull(path);
-    checkNotNull(payload);
-
-    if (assetKind != HELM_PACKAGE) {
-      throw new IllegalArgumentException("Unsupported assetKind: " + assetKind);
-    }
-
     try (TempBlob tempBlob = facet(StorageFacet.class).createTempBlob(payload, HASH_ALGORITHMS)) {
-      storeChart(path, tempBlob, payload);
+      upload(path, tempBlob, payload, assetKind);
     }
   }
 
   @Override
-  public Asset upload(String path, TempBlob tempBlob,Payload payload) throws IOException {
+  public Asset upload(String path, TempBlob tempBlob, Payload payload, AssetKind assetKind) throws IOException {
```

Comment:
This method is called from `UploadHandler` and transaction must not be opened there. Add `@TransactionalStoreBlob` here

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/84#discussion_r343583193